archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java

```diff
@@ -98,6 +103,12 @@
 public class ConfigProxyFactory {
     private static final Logger LOG = LoggerFactory.getLogger(ConfigProxyFactory.class);
 
+    // Users sometimes leak both factories and proxies, leading to memory problems that get blamed on us ;-)
+    // We'll use these maps to keep track of how many instances of each are created and make log noise to help them
```

Comment:
extra nit: code style: different javadoc style from the rest of the file

Reply:
Not really a javadoc, more an internal comment for implementors :-)